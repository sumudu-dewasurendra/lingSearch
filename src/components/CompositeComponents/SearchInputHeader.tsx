import React from 'react';
import {View, StyleSheet} from 'react-native';
import SearchInput from '../SearchInput';
import Button from '../Button';

const styles = StyleSheet.create({
  searchInputHeader: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
});

type SearchInputHeaderProps = {
  searchText: string;
  onChangeSearchText: (text: string) => void;
  searchButtonLabel: string;
  onPresSearchButton: () => void;
  searchButtonWidth?: number;
  searchButtonHeight?: number;
  searchbuttonBorderRadius?: number;
  searchButtonWrapperStyles: {[key: string]: any};
  searchButtonTextColor?: string;
};

const SearchInputHeader = ({
  searchText,
  onChangeSearchText,
  searchButtonLabel,
  onPresSearchButton,
  searchbuttonBorderRadius,
  searchButtonWrapperStyles,
  searchButtonTextColor,
}: SearchInputHeaderProps) => {
  return (
    <View style={styles.searchInputHeader}>
      <SearchInput
        searchText={searchText}
        onChangeSearchText={onChangeSearchText}
      />
      <Button
        label={searchButtonLabel}
        onPressButton={onPresSearchButton}
        borderRadius={searchbuttonBorderRadius}
        wrapperStyles={searchButtonWrapperStyles}
        labelColor={searchButtonTextColor}
      />
    </View>
  );
};

export default SearchInputHeader;
