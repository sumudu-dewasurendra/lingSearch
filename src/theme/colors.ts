export const COLORS = {
  BACKGROUND_COLOR: '#fff9bd',
  BORDER_COLOR: '#544d03',
  TEXT_COLOR: '#000',
};
