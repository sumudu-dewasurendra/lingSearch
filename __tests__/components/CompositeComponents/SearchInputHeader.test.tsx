import React from 'react';
import {create} from 'react-test-renderer';
import SearchInputHeader from '../../../src/components/CompositeComponents/SearchInputHeader';
import {COLORS} from '../../../src/theme/colors';
import {expect, test} from '@jest/globals';

test('renders SearchInputHeader component correctly', () => {
  const tree = create(
    <SearchInputHeader
      searchText={''}
      onChangeSearchText={(text: string) => console.log(text)}
      searchButtonLabel="Search"
      onPresSearchButton={() => console.log('onPresSearchButton')}
      searchbuttonBorderRadius={30}
      searchButtonWrapperStyles={{}}
      searchButtonTextColor={COLORS.TEXT_COLOR_WHITE}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
