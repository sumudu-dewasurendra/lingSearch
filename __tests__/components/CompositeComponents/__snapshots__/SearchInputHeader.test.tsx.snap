// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchInputHeader component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f9f9f9",
        "borderColor": "#c5c5af",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "width": "70%",
      }
    }
  >
    <Text>
      🔍
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Search by name"
      style={
        {
          "padding": 16,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "#c5c5af",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 5,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "700",
          "letterSpacing": 0.5,
        }
      }
    >
      Search
    </Text>
  </View>
</View>
`;
